test/extended/prometheus/prometheus.go

```diff
@@ -310,7 +310,11 @@ func bringUpPrometheusFromTemplate(oc *exutil.CLI) (ns, host, bearerToken string
 		stdout, _, err := oc.WithoutNamespace().Run("process").Args("-f", configPath).Outputs()
 		o.Expect(err).NotTo(o.HaveOccurred())
 		err = oc.WithoutNamespace().AsAdmin().Run("create").Args("-f", "-").InputString(stdout).Execute()
-		o.Expect(err).NotTo(o.HaveOccurred())
+		// still check if it exists, as the prior not found check above may have been in a race with a test
+		// running in another process
+		if !kapierrs.IsAlreadyExists(err) {
```

Comment:
you're not making an api call here, you're effectively shelling out to invoke "oc create -f", so this error check is not going to work.

Reply:
you may have to literally scan the output from the command.  Or just log the error w/o failing the test (so if the test ultimately fails, we can at least go back and discover the error that occurred)

Reply:
I'll go with the latter ... the existing use of `IsNotFound` presumably also falls into this bucket

Reply:
The existing call is not using the oc client:
```
if _, err := oc.AdminKubeClient().Apps().StatefulSets("kube-system").Get("prometheus", metav1.GetOptions{}); err != nil {
if !kapierrs.IsNotFound(err) {
```

it's using the KubeClient (aka the API client)

so it's fine.

You could consider doing the same thing here.

Reply:
I've been exploring the various client entryways in cli.go but am not finding a verb that allows me to "process" the template

I'll look a bit more but unless you have a precise pointer, thinking of falling back to parsing output.

And thanks for the clarification around the not found path.

Reply:
you don't have to change the process logic, just the create logic.

Reply:
but honestly this should fail so rarely i'd be fine w/ you just logging the error so if the test fails further down, we have the error on record.